static/src/javascripts/projects/common/modules/navigation/membership.js

```diff
@@ -50,11 +50,11 @@ const showAccountDataUpdateWarningMessage = accountDataUpdateWarningLink => {
         )}'>update your details</a>`
     );
 };
+const updateLink = accountDataUpdateWarning();
 
-const canShow = () => Promise.resolve(accountDataUpdateWarning() !== null);
+const canShow = () => Promise.resolve(updateLink !== null);
 
```

Comment:
I really don't like that. If `show()` gets called without running `canShow()` first for any reason you end up with null

Reply:
OK I updated above from `let updateLink;` to `let updateLink = null`. So if for whatever reason `show()` is called without running `canShow()` (it should never really happen, but yes it is possible)...then the conditional in `show()` won't pass and the banner won't be shown, which I feel is the most desirable behaviour as `updateLink` being `null` is an indicator that the required checks either haven't taken place, or they have and they've not returned a pass condition.

Reply:
This is a minor bit but I've been thinking about eventually having a `#display-message={message-id}` feature, both for development and for rare cases where you would want to force a message and that would break. (For example, i got a request for this for the sign in alert, which has  feedback form)

Regardless of these examples I really feel it's extremely desirable to keep the code separated. `canShow()` only runs checks, `show()` only shows the banner no matter what. Having a global side effect in `canShow` that can break `show` just doesn't sit right with me